tool/tctl/common/plugin/awsic.go

```diff
@@ -0,0 +1,242 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	pluginspb "github.com/gravitational/teleport/api/gen/proto/go/teleport/plugins/v1"
+	"github.com/gravitational/teleport/api/types"
+	apicommon "github.com/gravitational/teleport/api/types/common"
+	"github.com/gravitational/teleport/lib/utils"
+	awsutils "github.com/gravitational/teleport/lib/utils/aws"
+	toolcommon "github.com/gravitational/teleport/tool/common"
+)
+
+type awsICArgs struct {
+	cmd                  *kingpin.CmdClause
+	defaultOwners        []string
+	scimToken            string
+	scimURL              *url.URL
+	region               string
+	arn                  string
+	useSystemCredentials bool
+	userOrigins          []string
+	userLabels           toolcommon.LabelFilters
+	groupNameFilters     []string
+	accountNameFilters   []string
+	accountIDFilters     []string
+}
+
+func (a *awsICArgs) validate() error {
+	if !awsutils.IsKnownRegion(a.region) {
+		return trace.BadParameter("unknown AWS region: %s", a.region)
+	}
+
+	if a.scimToken == "" {
+		return trace.BadParameter("SCIM token must not be empty")
+	}
+
+	if !a.useSystemCredentials {
+		return trace.BadParameter("only AWS Local system credentials are supported")
+	}
+
+	return nil
+}
+
+// parseAWSICNameFilters validates that all elements of the supplied [names] slice
+// are valid regexes or globs and wraps them in [types.AWSICResourceFilter]s for
+// inclusion in a [types.PluginAWSICSettings].
+//
+// We are using a manual validator here rather than the canonical one defined
+// in the AWS IC integration itself, because those filter tools are not
+// available to OSS builds of tctl.
+//
+// TODO(tcsc): Move the filter validation somewhere that the that the OSS build can access it.
```

Comment:
This is OSS code. Is the TODO still relevant?

Reply:
Yes, but it could be clearer that I mean the validation code in `e`, rather than the copy of it I have here.